src/unused_finder/export_collector.rs

```diff
@@ -0,0 +1,477 @@
+                        // export { foo } from 'foo'
+                        if !self
+                            .skipped_items
+                            .iter()
+                            .any(|skipped| skipped.is_match(&ident.sym.to_string()))
+                        {
+                            return Some(ImportedItem::Named(ident.sym.to_string()));
+                        }
+                    }
+                }
+                return None;
+            })
+            .collect();
+        if let Some(entry) = self.export_from_ids.get_mut(&source.value.to_string()) {
+            specifiers.drain(0..).for_each(|s| {
+                entry.insert(s);
+            })
+        } else {
+            self.export_from_ids
+                .insert(source.value.to_string(), HashSet::from_iter(specifiers));
+        }
+    }
+
+    /**
+     * Extracts information from the ExportSpecifier to construct the map of exported items with its values.
+     * supports `export { foo }` and aliased `export { foo as bar }`
+     */
+    fn handle_export_named_specifiers(&mut self, specs: &Vec<ExportSpecifier>) {
+        specs.iter().for_each(|specifier| match specifier {
+            ExportSpecifier::Named(named) => {
+                // Handles `export { foo as bar }`
+                if let Some(exported) = &named.exported {
+                    if let ModuleExportName::Ident(id) = exported {
+                        let sym = id.sym.to_string();
+                        // export { foo as default }
+                        if sym == "default" {
+                            self.exported_ids.insert(ExportedItem::Default);
+                        } else {
+                            if !self
+                                .skipped_items
+                                .iter()
+                                .any(|skipped| skipped.is_match(&sym))
+                            {
+                                self.exported_ids
+                                    .insert(ExportedItem::Named(id.sym.to_string()));
+                            }
+                        }
+                    }
+                } else if let ModuleExportName::Ident(id) = &named.orig {
+                    // handles `export { foo }`
+                    if !self
+                        .skipped_items
+                        .iter()
+                        .any(|skipped| skipped.is_match(&id.sym.to_string()))
+                    {
+                        self.exported_ids
+                            .insert(ExportedItem::Named(id.sym.to_string()));
+                    }
+                }
+            }
+            _ => {}
+        });
+    }
+
+    pub fn has_disable_export_comment(&self, lo: BytePos) -> bool {
+        if let Some(comments) = self.comments.get_leading(lo) {
+            return comments
+                .iter()
+                .any(|c| c.kind == CommentKind::Line && c.text.trim().eq("@ALLOW-UNUSED-EXPORT"));
```

Comment:
check string prefix instead of comment content -- should allow explanations as suffix
// @ALLOW-UNUSED-EXPORT <reason>